core/src/com/unciv/models/UnitAction.kt

```diff
@@ -1,12 +1,11 @@
 package com.unciv.models
 
 data class UnitAction(
-        var type: UnitActionType,
-        var canAct: Boolean,
-        var title: String = type.value,
-        var isCurrentAction: Boolean = false,
-        var uncivSound: UncivSound = UncivSound.Click,
-        var action: (() -> Unit)? = null
+        val type: UnitActionType,
+        val title: String = type.value,
+        val isCurrentAction: Boolean = false,
+        val uncivSound: UncivSound = UncivSound.Click,
+        val action: (() -> Unit)? = null
```

Comment:
Here I got rid of canAct since it was redundant with null actions.  So now you can just use null actions to indicate that you cannot act and any action means that you can act.  an action that is disabled is useless anyway